### PR TITLE
Adds get_coordinate_frame/ get_channel_properties

### PR DESCRIPTION
**Pull request by lrodri29**
Adds `get_coordinate_frame()` and `get_experiment()`
Both are requested functions. 
Make use of the get_project function to simplify grabbing an ExperimentResource and a CoordinateFrameResource. 

Below is a sample script:
```python
import intern
from intern.remote.boss import BossRemote

rmt = BossRemote({
    "protocol": "https",
    "host": "api.theboss.io",
    "token": "TOKEN"
})

# params
col = 'kasthuri2015'
exp = 'em'
chan = 'cc'
initial_res = 0

#Bounds
x_bounds = [6400,6400+100]
y_bounds = [9000,9000+100]
z_bounds = [843, 848]

#To get base resolution
experiment = rmt.get_experiment(col, exp)
coordinate_frame = experiment.coord_frame

# To get voxel_sizes
cf = rmt.get_coordinate_frame(coordinate_frame)
x_voxel_size = cf.x_voxel_size
```

### CODE REVIEW

File: intern/remote/boss/remote.py [hunk 1]
```diff
@@ -903,3 +903,27 @@ def get_cutout(self, resource, resolution, x_range, y_range, z_range, time_range
...
+
+    def get_channel_properties(self, resource):
+        """
+        Gets a json dictionary of the channel properties displayed on the boss
+        
+        Args:
+            resource (intern.resource.Resource): A channel.
+
+        Returns:
+            (dict): {'base_resolution': 0, 'default_time_sample': 0, 'downsample_status': 'DOWNSAMPLED', 'experiment': 'em', 'datatype': 'uint8', 'sources': [], 'description': '', 'name': 'cc', 'creator': 'will', 'related': ['syn_obj-test2', 'syn_obj_v1'], 'type': 'image'}
```
Comment (j6k4m8) [inline intern/remote/boss/remote.py]: Can we replace without specific example data?

File: intern/remote/boss/remote.py [hunk 2]
```diff
@@ -903,3 +903,27 @@ def get_cutout(self, resource, resolution, x_range, y_range, z_range, time_range
...
+
+    def get_coordinate_frame(self, name):
+        """
+        Gets the coordinate frame properties displayed under the coordinate frame on the boss
+
+        Args:
+            name (str): Name of the coordinate frame
+        
+        Returns:
+            (dict) : {'voxel_unit': 'nanometers', 'z_start': 0, 'x_start': 0, 'z_voxel_size': 30.0, 'x_voxel_size': 6.0, 'x_stop': 10752, 'y_start': 0, 'name': 'kasthuri2015_6nm_offset0', 'z_stop': 1849, 'description': '6nm data', 'y_stop': 13312, 'y_voxel_size': 6.0}
```
Comment (j6k4m8) [inline intern/remote/boss/remote.py]: Same here

**Comment by j6k4m8** [comment]
Luis hmu tomorrow, I have code for you!

File: intern/remote/boss/remote.py (hunk 2 above)
Comment (lrodri29) [reply intern/remote/boss/remote.py]: Done! :)

**Comment by lrodri29** [comment]
Thank you all, let me know if anything else needs changing

File: intern/service/boss/baseversion.py [hunk 3]
```diff
@@ -15,6 +15,7 @@
 from abc import ABCMeta, abstractmethod
 from intern.resource.boss.resource import CoordinateFrameResource
 from requests import Request
+from requests import get
```
Comment (movestill) [inline intern/service/boss/baseversion.py]: This is no longer needed, right?

File: intern/remote/boss/remote.py [hunk 4]
```diff
@@ -903,3 +903,29 @@ def get_cutout(self, resource, resolution, x_range, y_range, z_range, time_range
...
+
+    def get_experiment(self, coll_name, exp_name):
+        """
+        Gets a json dictionary of the channel properties displayed on the boss
+        
+        Args:
+            resource (intern.resource.Resource): A channel.
+
+        Returns:
+            (dict): Dictionary of channel properties
```
Comment (movestill) [inline intern/remote/boss/remote.py]: Return type will be `ExperimentResource`.

File: intern/remote/boss/remote.py [hunk 5]
```diff
@@ -903,3 +903,29 @@ def get_cutout(self, resource, resolution, x_range, y_range, z_range, time_range
...
+
+    def get_coordinate_frame(self, name):
+        """
+        Gets the coordinate frame properties displayed under the coordinate frame on the boss
+
+        Args:
+            name (str): Name of the coordinate frame
+        
+        Returns:
+            (dictionary): Dictionary of coordinate frame information.
```
Comment (movestill) [inline intern/remote/boss/remote.py]: Return type will be `CoordinateFrameResource`.

File: intern/remote/boss/remote.py [hunk 6]
```diff
@@ -924,6 +924,6 @@ def get_coordinate_frame(self, name):
             name (str): Name of the coordinate frame
         
         Returns:
-            (dict) : {'voxel_unit': 'nanometers', 'z_start': 0, 'x_start': 0, 'z_voxel_size': 30.0, 'x_voxel_size': 6.0, 'x_stop': 10752, 'y_start': 0, 'name': 'kasthuri2015_6nm_offset0', 'z_stop': 1849, 'description': '6nm data', 'y_stop': 13312, 'y_voxel_size': 6.0}
+            (dicttionary): Coordinate frame in a dictionary.
```
Comment (j6k4m8) [inline intern/remote/boss/remote.py]: This returns a `CoordinateFrameResource` now, right?

File: intern/service/boss/project.py [hunk 7]
```diff
@@ -463,6 +463,18 @@ def delete(self, resource):
...
 
+    def get_channel_properties(self, resource):
+        """
+        Gets a json dictionary of the channel properties displayed on the boss
+        
+        Args:
+            resource (intern.resource.Resource): A channel.
+
+        Returns:
+            (dict): {'base_resolution': 0, 'default_time_sample': 0, 'downsample_status': 'DOWNSAMPLED', 'experiment': 'em', 'datatype': 'uint8', 'sources': [], 'description': '', 'name': 'cc', 'creator': 'will', 'related': ['syn_obj-test2', 'syn_obj_v1'], 'type': 'image'}
```
Comment (j6k4m8) [inline intern/service/boss/project.py]: Same here, probably just list as CFResource

File: intern/service/boss/project.py [hunk 8]
```diff
@@ -471,6 +483,6 @@ def get_coordinate_frame(self, name):
             name (str): Name of the coordinate frame
         
         Returns:
-            (dict) : {'voxel_unit': 'nanometers', 'z_start': 0, 'x_start': 0, 'z_voxel_size': 30.0, 'x_voxel_size': 6.0, 'x_stop': 10752, 'y_start': 0, 'name': 'kasthuri2015_6nm_offset0', 'z_stop': 1849, 'description': '6nm data', 'y_stop': 13312, 'y_voxel_size': 6.0}
+            (dictionary) : Coordinate frame encoded in a dictionary
```
Comment (j6k4m8) [inline intern/service/boss/project.py]: ditto :)

File: intern/service/boss/baseversion.py (hunk 3 above)
Comment (lrodri29) [reply intern/service/boss/baseversion.py]: Yeah was done in Clean Up commit haha- I guess you started your review before I committed that

File: intern/remote/boss/remote.py [hunk 9]
```diff
@@ -906,26 +906,27 @@ def get_cutout(self, resource, resolution, x_range, y_range, z_range, time_range
...
+        Convenience method that gets experiment resource.
         
         Args:
-            resource (intern.resource.Resource): A channel.
+            coll_name (str): Collection name
+            exp_name (str): Experiment name
 
         Returns:
-            (dict): Dictionary of channel properties
+            ExperimentResource
```
Comment (movestill) [inline intern/remote/boss/remote.py]: Enclose in parantheses for consistency with the rest of the docstrings.

File: intern/remote/boss/remote.py [hunk 10]
```diff
@@ -906,26 +906,27 @@ def get_cutout(self, resource, resolution, x_range, y_range, z_range, time_range
...
         """
-        Gets the coordinate frame properties displayed under the coordinate frame on the boss
+        Convenience method that gets coordinate frame resource
 
         Args:
             name (str): Name of the coordinate frame
         
         Returns:
-            (dictionary): Dictionary of coordinate frame information.
+            CoordinateFrame Resource
```
Comment (movestill) [inline intern/remote/boss/remote.py]: Enclose in parantheses for consistency with the rest of the docstrings and remove space.

**Comment by j6k4m8** [comment]
Also don't forget to add these to the changelog :)

**Comment by j6k4m8** [comment]
(resolved issue in changelog)

**Comment by lrodri29** [comment]
Is this good to go? Not sure what changes are required

**Pull request merged**